### PR TITLE
Append test expansion

### PR DESCRIPTION
**Pull request by jonkeane**
Explicit order checking, and details about reproduction of the transient bug.

### CODE REVIEW

File: tests/testthat/test-merge-datasets.R [hunk 1]
```diff
@@ -183,6 +183,11 @@ with_test_authentication({
     })
 
     test_that("Similarly uncomplicated merge, but numeric and hidden key", {
+        # When merging datasets during jenkins run tests, the order of the
+        # variables is *sometiems* not copied (usually Pet Name/q3 is missing),
```
Comment (bertjwregeer) [inline tests/testthat/test-merge-datasets.R]: s/sometiems/sometimes/

File: tests/testthat/test-merge-datasets.R [hunk 2]
```diff
@@ -227,16 +232,71 @@ with_test_authentication({
...
+                          collapse="\n"),
+                      fixed=TRUE)
         expect_identical(names(ds1),
             c("id", "matches", "other_var", "q1", "petloc"))
         expect_equal(sum(table(ds1$q1)), 1)
     })
 
     test_that("Can select rows to join", {
+        # When merging datasets during jenkins run tests, the order of the
+        # variables is *sometiems* not copied (usually Pet Name/q3 is missing),
```
Comment (bertjwregeer) [inline tests/testthat/test-merge-datasets.R]: s/sometiems/sometimes/

File: tests/testthat/test-merge-datasets.R [hunk 3]
```diff
@@ -227,16 +232,71 @@ with_test_authentication({
...
+                          paste0("[+] ", name(ds2)),
+                          paste("    ",
+                                c(
+                                    "[+] Key Pet Indicators",
+                                    "    Pet",
+                                    "    Pets by location",
+                                    "[+] Dog Metrics",
+                                    "    [+] Number of dogs by type",
+                                    "        (Empty group)",
+                                    "[+] Details",
```
Comment (nealrichardson) [inline tests/testthat/test-merge-datasets.R]: We're copying over empty groups? That seems undesirable.

**Comment by nealrichardson** [comment]
One comment but not about the test. Otherwise LGTM.

**Pull request merged**